internal/provider/common.go

```diff
@@ -140,10 +266,14 @@ func getUpdateParamsForPlatformBM(platform *models.Platform, userManagedNetworki
 func getUpdateParamsForPlatformUMNMandatory(platform *models.Platform, userManagedNetworking *bool, cluster *common.Cluster) (*models.Platform, *bool, error) {
 	if (userManagedNetworking == nil || *userManagedNetworking) && isUMNMandatoryForPlatform(platform) {
 		// userManagedNetworking is already set to true, nothing to do
-		if platform == nil || *cluster.Platform.Type == *platform.Type {
+		if platform == nil || platform.Type == nil {
+			return nil, nil, nil
+		} else if *cluster.Platform.Type == *platform.Type && *cluster.Platform.Type == models.PlatformTypeExternal {
+			return checkExternalPlatformUpdate(*platform, cluster), nil, nil
+		} else if *cluster.Platform.Type == *platform.Type {
 			return nil, nil, nil
 		} else {
```

Comment:
those conditions are returning early, wouldn't it be clearer if we remove the `else if/else` part?

Reply:
How do you think we can remove it?

Reply:
I will move it to a more generic function `checkPlatformUpdate`, it's required to no set the usages each time we update the platform type. I really don't know if we need to keep that behavior as it makes things complicated, but I'll let it like that for the moment.

Reply:
```suggestion
	if platform == nil || platform.Type == nil {
		return nil, nil, nil
	}
        if *cluster.Platform.Type == *platform.Type && *cluster.Platform.Type == models.PlatformTypeExternal {
		return checkExternalPlatformUpdate(*platform, cluster), nil, nil
	}
	if *cluster.Platform.Type == *platform.Type {
		return nil, nil, nil
	}
	return updatePlatformIsExternal(platform), nil, nil

```

Reply:
updated !